pages/api/internal-jobs/email-digest/index.ts

```diff
@@ -0,0 +1,270 @@
+      logger.warn('Skipping user ' + user.name + ' because they have already received an email digest in the past 24 hours');
+      continue;
+    }
+
+    if (totalEmailedSoFar.includes(user.email)) {
+      logger.warn('Skipping user ' + user.name + ' because they have already received an email digest in this batch');
+      continue;
+    }
+
+    logger.warn('Sending email to user ' + user.name + ' (' + user.email + ')');
+
+    const todaysDatePretty = new Date().toLocaleDateString('en-US', { year: 'numeric', month: 'long', day: 'numeric' });
+    const subject = userConfig.emailDigest === EmailDigestSettingTypes.DAILY ?
+      `Daily Digest - ${todaysDatePretty}` :
+      `You have ${notificationsCount} new notification${notificationsCount !== 1 ? 's' : ''}`;
+
+    const title = `Welcome to the Pathology daily digest for ${todaysDatePretty}.`;
+    const body = getEmailBody(levelOfDay, notificationsCount, title, user);
+    const sentError = await sendMail(batchId, EmailType.EMAIL_DIGEST, user, subject, body);
+
+    if (!sentError) {
+      sentList.push(user.email);
+    }
+    else {
+      failedList.push(user.email);
+    }
+  }
+
+  return { sentList, failedList };
+}
+
+export async function sendEmailReactivation(batchId: ObjectId, totalEmailedSoFar: string[] = []) {
+  // if they haven't been active in 7 days and they have an email address, send them an email, but only once every 90 days
+  // get users that haven't been active in 7 days
+  const levelOfDay = await getLevelOfDay();
+  const usersThatHaveBeenSentReactivationInPast30d = await EmailLogModel.find({
+    type: EmailType.EMAIL_7D_REACTIVATE,
+    createdAt: { $gte: new Date(Date.now() - 90 * 24 * 60 * 60 * 1000) },
+  }).distinct('userId');
+
+  const inactiveUsers = await UserModel.aggregate([
+    {
+      $match: {
+        _id: { $nin: usersThatHaveBeenSentReactivationInPast30d },
+
+        last_visited_at: { $lte: (Date.now() / 1000) - (7 * 24 * 60 * 60 ) }, // TODO need to refactor last_visited_at to be a DATE object instead of seconds
+
+        email: { $ne: null },
+      },
+    },
+    {
+      $lookup: {
+        from: 'userconfigs',
+        localField: '_id',
+        foreignField: 'userId',
+        as: 'userConfig',
+      },
+    },
+    {
+      $unwind: {
+        path: '$userConfig',
+        preserveNullAndEmptyArrays: false, // if user has no config, don't include them
+      },
+    },
+    {
+      $match: {
+        'userConfig.emailDigest': { $ne: EmailDigestSettingTypes.NONE },
+      },
+    },
+    {
+      $project: {
+        _id: 1,
+        name: 1,
+        email: 1,
+        userConfig: 1,
+        score: 1
+      }
+    }
+  ]);
+
+  const sentList: string[] = [];
+  const failedList: string[] = [];
+  const totalLevels = await LevelModel.countDocuments({
+    isDraft: false,
+  });
+  const totalCreators = await LevelModel.distinct('userId').countDocuments();
+
+  for (const user of inactiveUsers) {
```

Comment:
should we check each userConfig for `EmailDigestSettingTypes.NONE` before sending?

Reply:
The aggregate query above should account for this